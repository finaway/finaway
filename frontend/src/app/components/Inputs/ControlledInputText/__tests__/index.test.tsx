import * as React from 'react';
import { render, renderHook } from '@testing-library/react';

import { ControlledInputText } from '..';
import { useForm } from 'react-hook-form';

interface FormValues {
  test: string;
}

describe('<ControlledInputText />', () => {
  const { result } = renderHook(() => useForm<FormValues>());
  const { control } = result.current;

  it('should match snapshot', () => {
    const loadingIndicator = render(
      <ControlledInputText control={control} name="test" label="Input Test" />,
    );
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });

  it('input should be disabled', () => {
    const { getByTestId } = render(
      <ControlledInputText
        control={control}
        name="test"
        label="Input Test"
        disabled
        data-testid="input-test"
      />,
    );
    expect(getByTestId('input-test')).toBeDisabled();
  });

  it('input should be autofocus', () => {
    const { getByTestId } = render(
      <ControlledInputText
        control={control}
        name="test"
        label="Input Test"
        data-testid="input-test"
        autoFocus
      />,
    );
    expect(getByTestId('input-test')).toHaveFocus();
  });

  it('input should be type password', () => {
    const { getByTestId } = render(
      <ControlledInputText
        control={control}
        name="test"
        label="Input Test"
        data-testid="input-test"
        type="password"
      />,
    );
    expect(getByTestId('input-test')).toHaveAttribute('type', 'password');
  });

  it('input label should be Input Test', () => {
    const red = render(
      <ControlledInputText
        control={control}
        name="test"
        label="Input Test"
        data-testid="input-test"
      />,
    );

    expect(red.getByTestId('input-test').parentNode).toHaveTextContent(
      'Input Test',
    );
  });
});
