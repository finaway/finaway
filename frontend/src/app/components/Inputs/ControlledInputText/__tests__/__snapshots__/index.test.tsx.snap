// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlledInputText /> should match snapshot 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink="false"
    for=""
  >
    Input Test
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
  >
    <input
      aria-invalid="false"
      autocomplete="off"
      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid=""
      id=""
      name="test"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-yjsfm1"
      >
        <span>
          Input Test
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
